pkg/datapath/iptables/iptables.go

```diff
@@ -574,11 +574,23 @@ func (m *IptablesManager) installStaticProxyRules() error {
 				"-j", "ACCEPT"), false)
 		}
 		if err == nil {
-			// No conntrack for proxy return traffic
+			// No conntrack for proxy return traffic that is heading to lxc+
+			err = runProg("iptables", append(
+				m.waitArgs,
+				"-t", "raw",
+				"-A", ciliumOutputRawChain,
+				"-o", "lxc+",
+				"-m", "mark", "--mark", matchProxyReply,
+				"-m", "comment", "--comment", "cilium: NOTRACK for proxy return traffic",
+				"-j", "NOTRACK"), false)
+		}
```

Comment:
Interesting, OK, so overall we're basically reducing the scope of this rule to only apply to traffic going towards the endpoints. The change below to add `-o cilium_host` supports this for the regular mode, then this rule supports this for `--enable-endpoint-routes` where the packets destined for a pod are routed directly to the `lxc` device. Should we gate this particular rule insertion by `--enable-endpoint-routes` mode?

Reply:
Given that we want to move towards enabling endpoint routes by default, I'm OK to add it unconditionally. `installForwardChainRules()` also adds these rules unconditionally.